addons/mrp/models/stock_move.py

```diff
@@ -309,3 +338,16 @@ def _compute_kit_quantities(self, product_id, kit_qty, kit_bom, filters):
             return min(qty_ratios) // 1
         else:
             return 0.0
+
+    def _decrease_initial_demand(self, qty, stream='UP'):
+        done_move_to_return = []
+        finished_moves = self.env['stock.move']
+        for move in self:
+            if move.production_id and move.state not in ('done, cancel'):
+                try:
+                    if move.production_id.state == 'confirmed':
+                        move.production_id._update_quantity(move.production_id.product_qty - qty)
```

Comment:
huh? qty is the quantity decreased on the move and you want to decrease this quantity to the qty to produce of a mo? if it's the case then it doesn't make sens, you'll have to take into account the unit factor, uom.. and i'm not even sure it does make sense (it'll update all the other raw moves?)

Reply:
It concerned the finished move but you are right I need to make the uom conversion